### PR TITLE
Pass all kwargs to celery task when retrying

### PR DESCRIPTION
**Pull request by majamassarini**
I found no way to unit test it (suggestions are welcome)

Has been tested with docker compose but using another exception as trigger.
I can not reach this code, as it is, with docker compose because of multiple reasons:

1. tokman pod is lacking
2. fedpkg seems not to be able to access the repository

Should fix packit-service#1426

---

RELEASE NOTES BEGIN
RELEASE NOTES END

### CODE REVIEW

File: packit_service/worker/handlers/distgit.py [hunk 1]
```diff
@@ -171,13 +171,10 @@ def sync_branch(
                     logger.info(f"Will retry for the {retries + 1}. time in {delay}s.")
                     # throw=False so that exception is not raised and task
                     # is not retried also automatically
+                    kargs = self.task.request.kwargs.copy()
```
Comment (nikromen) [inline packit_service/worker/handlers/distgit.py:174]: I think you ment `kwargs` not `kargs` here :D
otherwise this should work :pray:

**Review by nikromen — APPROVED** [review]

**Review by nikromen — COMMENTED** [review]
I am just wondering - to make debugging easier if something will again happen to this - can we add some logs -> with which arguments are we retrying the task?

File: packit_service/worker/handlers/distgit.py (hunk 1 above)
Comment (majamassarini) [reply packit_service/worker/handlers/distgit.py:174]: No, I did not use kwargs as the variable name for purpose but I want a name that recalls it (and it seems to me that it worked :-). I do not want to lead to potential future bugs using a standard keyword name for a different purpose.

File: packit_service/worker/handlers/distgit.py [hunk 2]
```diff
@@ -171,13 +171,10 @@ def sync_branch(
...
+                    kargs["propose_downstream_run_id"] = model.id
                     self.task.retry(
-                        exc=ex,
-                        countdown=delay,
-                        throw=False,
-                        args=(),
-                        # https://stackoverflow.com/questions/68915407/celery-retry-with-updated-arguments
```
[the remark this answers is not in the document]
Comment (majamassarini) [reply packit_service/worker/handlers/distgit.py:179]: I found it somehow misleading and this is the reason why I removed it. It says

> You may also opt to do the other way around where you would use kwargs={<values here>} and empty out args=()

But this is not always true, it depends on your task implementation.

I think that the best documentation is the one from the celery project: https://celeryproject.readthedocs.io/zh_CN/latest/userguide/tasks.html#retrying 

But I don't think we need to put it in the code :)

[the remark this answers is not in the document]
Comment (lbarcziova) [reply packit_service/worker/handlers/distgit.py:179]: > I think that the best documentation is the one from the celery project: https://celeryproject.readthedocs.io/zh_CN/latest/userguide/tasks.html#retrying

+1

I would even add it as a comment :)

**Review by lbarcziova — APPROVED** [review]
Thanks!

**Pull request merged**